port/esp32_port.c

```diff
@@ -142,18 +142,18 @@ esp_loader_error_t loader_port_read(uint8_t *data, uint16_t size, uint32_t timeo
 // assert reset pin for 50 milliseconds.
 void loader_port_enter_bootloader(void)
 {
-    gpio_set_level(s_gpio0_trigger_pin, 0);
+    gpio_set_level(s_gpio0_trigger_pin, SERIAL_FLASHER_BOOT_INVERT ? 1 : 0);
     loader_port_reset_target();
     loader_port_delay_ms(SERIAL_FLASHER_BOOT_HOLD_TIME_MS);
-    gpio_set_level(s_gpio0_trigger_pin, 1);
+    gpio_set_level(s_gpio0_trigger_pin, SERIAL_FLASHER_BOOT_INVERT ? 0 : 1);
 }
 
 
 void loader_port_reset_target(void)
 {
-    gpio_set_level(s_reset_trigger_pin, 0);
+    gpio_set_level(s_reset_trigger_pin, SERIAL_FLASHER_RESET_INVERT ? 1 : 0);
```

Comment:
```suggestion
    gpio_set_level(s_reset_trigger_pin, CONFIG_SERIAL_FLASHER_RESET_ACTIVE_LEVEL);
```

Reply:
I actually believe the initial approach is more readable and future-proof.